File: src/main/java/com/sjh14o3/mediator/bone/Colleague.java
package com.sjh14o3.mediator.bone;

public abstract class Colleague {
    protected final Mediator mediator;

    public abstract void transmitMessage(String message, Colleague target);

    public void receiveMessage(String message) {
        System.out.println("message \"" + message  + "\" received");
    }

    public Colleague(Mediator mediator) {
        this.mediator = mediator;
    }
}

File: src/main/java/com/sjh14o3/mediator/chatroom/Colleague.kt
package com.sjh14o3.mediator.chatroom

abstract class Colleague(mediator: Mediator) {
    protected val _mediator = mediator
    val mediator: Mediator
        get() = _mediator


    fun sendMessage(message: String) {
        mediator.sendMessage(message, this)
    }
    abstract fun receive(message: String)
}